consensus/propagation/reactor.go

```diff
@@ -272,12 +272,34 @@ func (blockProp *Reactor) Prune(committedHeight int64) {
 	blockProp.pmtx.Lock()
 	defer blockProp.pmtx.Unlock()
 	blockProp.consensusHeight = committedHeight
+	blockProp.ResetRequestCounts()
+}
+
+func (blockProp *Reactor) SetConsensusRound(height int64, round int32) {
+	blockProp.pmtx.Lock()
+	defer blockProp.pmtx.Unlock()
+	blockProp.consensusRound = round
+	blockProp.ResetRequestCounts()
+	// todo: delete the old round data as its no longer relevant don't delete
+	// past round data if it has a POL
+}
+
+func (blockProp *Reactor) ResetRequestCounts() {
+	peers := blockProp.getPeers()
+	for _, p := range peers {
```

Comment:
I would add a nil check: `if p != nil {` because I've see this happen sometime where a connected peer has a nil state and end up panicking. The issue seems deep so I left a TODO and expecting to investigate it afterwards